pinot-core/src/main/java/org/apache/pinot/core/periodictask/BasePeriodicTask.java

```diff
@@ -29,21 +32,33 @@
 @ThreadSafe
 public abstract class BasePeriodicTask implements PeriodicTask {
   private static final Logger LOGGER = LoggerFactory.getLogger(BasePeriodicTask.class);
+  private static final String DEFAULT_REQUEST_ID = "auto";
 
   // Wait for at most 30 seconds while calling stop() for task to terminate
   private static final long MAX_PERIODIC_TASK_STOP_TIME_MILLIS = 30_000L;
 
   protected final String _taskName;
   protected final long _intervalInSeconds;
   protected final long _initialDelayInSeconds;
+  protected final ReentrantLock _runLock;
 
   private volatile boolean _started;
   private volatile boolean _running;
 
+  // Default properties that tasks may use during execution. This variable is private and does not have any get or set
+  // methods to prevent subclasses from gaining direct access to this variable. See run(Properties) method to see how
+  // properties are passed and used during task execution.
+  private Properties _activePeriodicTaskProperties;
```

Comment:
Make this a constant and rename it
```suggestion
  private static final Properties DEFAULT_PERIODIC_TASK_PROPERTIES;
  
  static {
    DEFAULT_PERIODIC_TASK_PROPERTIES = new Properties();
    DEFAULT_PERIODIC_TASK_PROPERTIES.put(PeriodicTask.PROPERTY_KEY_REQUEST_ID, DEFAULT_REQUEST_ID);
  }
```

Reply:
Done.